src/Microsoft.ML.Vision/ImageClassificationTrainer.cs

```diff
@@ -1509,5 +1509,9 @@ public void Dispose()
 
             _isDisposed = true;
         }
+
+        // Finalizer to clean-up remaining Tensor objects generated in TensorFlow C libraries
+        // not cleaned up by GC
+        ~ImageClassificationModelParameters() => Dispose();
```

Comment:
Congrats on this working out! The usability will be much better, plus cleaner code-wise. Otherwise we have to make many classes in ML․NET `IDisposible` plus teach users the cases where they need to call `Dispose()`.

Might want to run locally and watch the memory usage over the run to see that it's stable after the 1st model is created (in a run of 100 image models). I'd expect it will oscillate but the peaks will remain stable; though the GC can delay the call to the finalizer.

After you're sure the leak is properly handled, I'd remove the `IDisposible` and `Dispose()` in this class and have the finalizer directly do the work inside of the current `Dispose()`:

```c#
~ImageClassificationModelParameters() {
  if (_session?.graph != IntPtr.Zero)
  {
    _session.graph.Dispose();
  }
  if (_session != null && _session != IntPtr.Zero)
  {
    _session.close();
  }
}
```

Then start removing ML․NET's recently added `IDisposible` which were added to support the disposing of the TF models.